vm/devices/vmbus/vmbus_server/src/lib.rs

```diff
@@ -822,19 +861,22 @@ impl ServerTask {
             }
         }
 
-        let open_request = params.zip(open).map(|(params, result)| {
-            let (_, interrupt) = self.inner.open_channel(offer_id, &params);
-            let channel = self.inner.complete_open(offer_id, Some(result));
-            OpenRequest::new(
-                params.open_data,
-                interrupt,
-                self.server
-                    .get_version()
-                    .expect("must be connected")
-                    .feature_flags,
-                channel.flags,
-            )
-        });
+        let open_request = params
+            .zip(open)
+            .map(|(params, result)| -> anyhow::Result<_> {
+                let (_, interrupt) = self.inner.open_channel(offer_id, &params);
+                let channel = self.inner.complete_open(offer_id, Some(result))?;
+                Ok(OpenRequest::new(
+                    params.open_data,
+                    interrupt,
+                    self.server
+                        .get_version()
+                        .expect("must be connected")
+                        .feature_flags,
+                    channel.flags,
+                ))
+            })
+            .transpose()?;
```

Comment:
This failure path is also arguably problematic--in this case, the device will see an error, so it will think the channel restore failed. But channels.rs will already have completed the restore above.

In practice, this probably doesn't matter, because any restore failure is going to result in the VM failing to start. But I think we want to make sure we catch this condition, just in case someone decides to do a "try restore" at some point in the future for some weird case. So I'd suggest either eliminating this failure path (if possible, probably by preallocating the ports before calling into channels.rs), or remember that a restore failed and then fail the vmbus_server post_restore() operation.